provider/bidengine/pricing.go

```diff
@@ -90,9 +138,29 @@ func (fp scalePricing) CalculatePrice(_ context.Context, _ string, gspec *dtypes
 		memoryQuantity = memoryQuantity.Mul(groupCount)
 		memoryTotal = memoryTotal.Add(memoryQuantity)
 
-		storageQuantity := decimal.NewFromBigInt(group.Resources.Storage.Quantity.Val.BigInt(), 0)
-		storageQuantity = storageQuantity.Mul(groupCount)
-		storageTotal = storageTotal.Add(storageQuantity)
+		for _, storage := range group.Resources.Storage {
+			storageQuantity := decimal.NewFromBigInt(storage.Quantity.Val.BigInt(), 0)
+			storageQuantity = storageQuantity.Mul(groupCount)
+
+			storageClass := ""
+			attr := storage.Attributes.Find(sdl.StorageAttributePersistent)
+			if isPersistent, _ := attr.AsBool(); isPersistent {
+				attr = storage.Attributes.Find(sdl.StorageAttributeClass)
+				if class, set := attr.AsString(); set {
+					storageClass = class
+				}
+			}
+
+			total, exists := storageTotal[storageClass]
```

Comment:
so if `ifPersistent` is `false`, then `storageClass` is going to be the empty string. So obviously `exists` is false. Which means the code below that checks `if !exists` is going to return an error. Does this mean all non-persistent storage classes cause a failure in this code?

Reply:
fixed